server/test/services/applicant/ApplicantDataTest.java

```diff
@@ -24,40 +27,72 @@ public void hasPreferredLocale_onlyReturnsTrueIfPreferredLocaleIsSet() {
     ApplicantData data = new ApplicantData();
     assertThat(data.hasPreferredLocale()).isFalse();
 
-    data = new ApplicantData("{\"applicant\":{}}");
+    data = new ApplicantData("{\"applicant\":{}}", null);
     assertThat(data.hasPreferredLocale()).isFalse();
 
-    data = new ApplicantData(Optional.empty(), "{\"applicant\":{}}");
+    data = new ApplicantData(Optional.empty(), "{\"applicant\":{}}", null);
     assertThat(data.hasPreferredLocale()).isFalse();
 
-    data = new ApplicantData(Optional.of(Locale.FRENCH), "{\"applicant\":{}}");
+    data = new ApplicantData(Optional.of(Locale.FRENCH), "{\"applicant\":{}}", null);
     assertThat(data.hasPreferredLocale()).isTrue();
   }
 
+  private ApplicantData createNewApplicantData() {
+    ApplicantModel applicant = new ApplicantModel();
+
+    // TODO (#5503): See the big block comment in ApplicantData#getApplicantName.
+    // Once we stop checking the first name against the account email address,
+    // we can remove this.
+    AccountModel account = new AccountModel();
+    account.setEmailAddress("test@email.com");
+    account.save();
+    applicant.setAccount(account);
+
+    applicant.save();
+    return new ApplicantData(applicant);
+  }
+
   @Test
   public void getApplicantName_exists() {
-    ApplicantData data = new ApplicantData();
+    ApplicantData data = createNewApplicantData();
```

Comment:
suggestion (non-blocking): Do you want to add a test that values fall back to WKP when PAI values aren't available? Maybe that's overkill since it would basically just be testing that `.or` call.

Reply:
Oh wow! Glad you tried it.